### PR TITLE
refactor config options to follow mimir/loki's `role-X` pattern

### LINKED ISSUE #13: refactor from `role` config option to use a `role-X` pattern like Mimir and Loki
### Enhancement Proposal

refactor from `role` config option to use a `role-X` pattern like Mimir and Loki

### PR DESCRIPTION
**Pull request by PietroPasotti**
testing instructions:

```
juju deploy cos-lite --channel edge
# configure facade to present s3
juju deploy facade --channel edge
cat << EOF > ./s3.yaml
endpoint: provide-s3
app_data: |
  {
    "access-key": "mykey",
    "bucket": "BuckyMcBucket",
    "endpoint": "http://1.2.3.4:9000",
    "secret-key": "soverysecret"
  }
EOF
juju run facade/0 update --params ./s3.yaml 
rm ./s3.yaml

juju deploy tempo-coordinator-k8s-operator --channel edge tempo

charmcraft pack
jhack deploy  # deploy from local

jhack imatrix fill
```

At this point you should have:

image

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -54,25 +54,94 @@ resources:
...
+        Configure the application to run with all roles enabled. This is the default configuration. 
+        This is a meta-role that configures the application to enable all other roles. Under the hood, the `all` role
+        is mapped to Tempo's `scalable-single-binary`.
+             
+        Each Tempo application can only have exactly one `role-X` config set to `true` at any given time. 
+        So if you want to configure this app to run with a role other than `all`, remember to set `all` to `false`.
+        Invalid configurations will result in the component to stop working and the app to set blocked status.
+        
+        Note that for a tempo deployment as a whole to be consistent, each role 
+        (except the optional 'query-frontend') needs to be assigned to at least one worker node. If this condition
```
Comment (michaeldmitry) [inline charmcraft.yaml:70]: `metrics-generator` is the optional role

File: charmcraft.yaml [hunk 2]
```diff
@@ -54,25 +54,94 @@ resources:
...
+    role-querier:
+      type: boolean
+      default: false
+      description: |
+        Configure the application to run as this Tempo role. Each Tempo application can only have exactly one 
+        `role-X` config set to `true`. So if you want to enable a role, remember to set all other roles to `false`.
+        Invalid configurations will result in the component to stop working and the app to set blocked status.
+        
+        Note that for a tempo deployment as a whole to be consistent, each role 
+        (except the optional 'query-frontend') needs to be assigned to at least one worker node. If this condition
```
Comment (michaeldmitry) [inline charmcraft.yaml:82]: And here

File: charmcraft.yaml [hunk 3]
```diff
@@ -54,25 +54,94 @@ resources:
...
+    role-query-frontend:
+      type: boolean
+      default: false
+      description: |
+        Configure the application to run as this Tempo role. Each Tempo application can only have exactly one 
+        `role-X` config set to `true`. So if you want to enable a role, remember to set all other roles to `false`.
+        Invalid configurations will result in the component to stop working and the app to set blocked status.
+        
+        Note that for a tempo deployment as a whole to be consistent, each role 
+        (except the optional 'query-frontend') needs to be assigned to at least one worker node. If this condition
```
Comment (michaeldmitry) [inline charmcraft.yaml:94]: And here

File: charmcraft.yaml [hunk 4]
```diff
@@ -54,25 +54,94 @@ resources:
...
+    role-ingester:
+      type: boolean
+      default: false
+      description: |
+        Configure the application to run as this Tempo role. Each Tempo application can only have exactly one 
+        `role-X` config set to `true`. So if you want to enable a role, remember to set all other roles to `false`.
+        Invalid configurations will result in the component to stop working and the app to set blocked status.
+        
+        Note that for a tempo deployment as a whole to be consistent, each role 
+        (except the optional 'query-frontend') needs to be assigned to at least one worker node. If this condition
```
Comment (michaeldmitry) [inline charmcraft.yaml:106]: And here

File: charmcraft.yaml [hunk 5]
```diff
@@ -54,25 +54,94 @@ resources:
...
+    role-distributor:
+      type: boolean
+      default: false
+      description: |
+        Configure the application to run as this Tempo role. Each Tempo application can only have exactly one 
+        `role-X` config set to `true`. So if you want to enable a role, remember to set all other roles to `false`.
+        Invalid configurations will result in the component to stop working and the app to set blocked status.
+        
+        Note that for a tempo deployment as a whole to be consistent, each role 
+        (except the optional 'query-frontend') needs to be assigned to at least one worker node. If this condition
```
Comment (michaeldmitry) [inline charmcraft.yaml:118]: And here

File: charmcraft.yaml [hunk 6]
```diff
@@ -54,25 +54,94 @@ resources:
...
+    role-compactor:
+      type: boolean
+      default: false
+      description: |
+        Configure the application to run as this Tempo role. Each Tempo application can only have exactly one 
+        `role-X` config set to `true`. So if you want to enable a role, remember to set all other roles to `false`.
+        Invalid configurations will result in the component to stop working and the app to set blocked status.
+        
+        Note that for a tempo deployment as a whole to be consistent, each role 
+        (except the optional 'query-frontend') needs to be assigned to at least one worker node. If this condition
```
Comment (michaeldmitry) [inline charmcraft.yaml:130]: And here

File: charmcraft.yaml [hunk 7]
```diff
@@ -54,25 +54,94 @@ resources:
...
+    role-metrics-generator:
+      type: boolean
+      default: false
+      description: |
+        Configure the application to run as this Tempo role. Each Tempo application can only have exactly one 
+        `role-X` config set to `true`. So if you want to enable a role, remember to set all other roles to `false`.
+        Invalid configurations will result in the component to stop working and the app to set blocked status.
+        
+        Note that for a tempo deployment as a whole to be consistent, each role 
+        (except the optional 'query-frontend') needs to be assigned to at least one worker node. If this condition
```
Comment (michaeldmitry) [inline charmcraft.yaml:142]: And finally here :)

Comment (PietroPasotti) [reply charmcraft.yaml:142]: hahaha ok ok sorry, got me

**Review by mmkay — APPROVED** [review]

**Review by lucabello — APPROVED** [review]

**Pull request merged**